### PR TITLE
feat(value-node): update method, serialised and parent getters

### PR DESCRIPTION
**Pull request by ilyamore88**
Added ValueNode methods and getters with unit tests.

### CODE REVIEW

File: src/entities/ValueNode/ValueNode.spec.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { createValueNodeMock } from '../../utils/mocks/createValueNodeMock';
+import { BlockNode, createBlockNodeName } from '../BlockNode';
+import { createBlockNodeMock } from '../../utils/mocks/createBlockNodeMock';
+import { EditorDocument } from '../EditorDocument';
+
+describe('ValueNode', () => {
+  describe('.update()', () => {
+    it('should add new data field associated with this value node', () => {
```
Comment (neSpecc) [inline src/entities/ValueNode/ValueNode.spec.ts:8]: Are you sure that we need to allow adding new properties to ValueNode? 

Seems like ValueNode should only allow updating of existed value which was set through the Constructor. And disallow adding new properties later

File: src/entities/BlockNode/types/BlockNodeChildren.ts [hunk 2]
```diff
@@ -10,4 +10,4 @@ import { FormattingNode } from '../../FormattingNode';
  * The child is the ValueNode if the block node contains the data that will be fully replaced on a change event.
  * If the value is an array, it can contain formatting nodes or text nodes. Changes in these nodes works in a range.
  */
-export type BlockNodeChildren = Record<DataKey, ValueNode | (FormattingNode | TextNode)[]>;
+export type BlockNodeChildren = Record<DataKey, ValueNode[] | (FormattingNode | TextNode)[]>;
```
Comment (gohabereg) [inline src/entities/BlockNode/types/BlockNodeChildren.ts:13]: Why array?

File: src/entities/ValueNode/ValueNode.spec.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import { createBlockNodeMock } from '../../utils/mocks/createBlockNodeMock';
+import { EditorDocument } from '../EditorDocument';
+import { createValueNodeName } from './types';
+
+describe('ValueNode', () => {
+  describe('.update()', () => {
+    it('should update existing data field associated with this value node', () => {
+      // Arrange
+      const valueNode = createValueNodeMock({
+        name: createValueNodeName('longitude'),
```
Comment (gohabereg) [inline src/entities/ValueNode/ValueNode.spec.ts:12]: I don't think node itself needs to know it's DataKey

File: src/entities/ValueNode/index.ts [hunk 4]
```diff
@@ -21,11 +26,44 @@ export class ValueNode {
...
+  public update(value: unknown): void {
+    this.#value = value;
+  }
+
+  /**
+   * Returns serialized data associated with this value node.
+   */
+  public get serialized(): ValueNodeSerialized {
+    return {
+      name: this.#name,
```
Comment (gohabereg) [inline src/entities/ValueNode/index.ts:53]: It should return just value

File: src/entities/ValueNode/ValueNode.spec.ts (hunk 3 above)
Comment (gohabereg) [reply src/entities/ValueNode/ValueNode.spec.ts:12]: Not sure about a parent as well though

**Review by gohabereg — CHANGES_REQUESTED** [review]

File: src/utils/mocks/createValueNodeMock.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { ValueNode } from '../../entities/ValueNode';
+
+/**
+ * Creates a mock ValueNode instance.
+ *
+ * @param args - ValueNode constructor arguments.
+ * @param args.value - The value of this value node.
+ */
+export function createValueNodeMock({ value }: { value: unknown }): ValueNode {
```
Comment (gohabereg) [inline src/utils/mocks/createValueNodeMock.ts:9]: You can use ConstructorParameters interface here

**Review by neSpecc — APPROVED** [review]
ok

**Review by gohabereg — APPROVED** [review]

**Pull request merged**